src/libopensc/card-idprime.c

```diff
@@ -380,11 +574,13 @@ static int idprime_fill_prkey_info(list_t *list, idprime_object_t **entry, sc_pk
 	/* Do not specify the length -- it will be read from the FCI */
 	prkey_info->path.count = -1;
 
-	/* TODO figure out the IDs as the original driver? */
-	prkey_info->id.value[0] = ((*entry)->fd >> 8) & 0xff;
-	prkey_info->id.value[1] = (*entry)->fd & 0xff;
-	prkey_info->id.len = 2;
+	memcpy(prkey_info->id.value, (*entry)->key_id, 8);
```

Comment:
This should say `(*entry)->key_len` instead of the 8.

Reply:
fixed and squashed commits